### DISCUSSION TITLE
Difficulty Accessing Logo Path in Typst Extension Due to Escaped Underscore and YAML Structure Concerns

### DISCUSSION THREAD

**Discussion by produnis** [quarto-dev/quarto-cli#11364] [Q&A]
### Description

I’ve translated one of my LaTeX extensions to Typst, and while it mostly works, it took a lot of trial and error. My issue revolves around a logo file I export in the `_extensions` folder. When I try to access this file in my Typst template, the underscore in the path is escaped to `\_`.

I read that this problem can be solved with `object.replace("\\", "")`, which I tried implementing. However, I ended up needing two lines of code to handle the logo path correctly from the YAML header:

```
#let logo_path = "$logo$"
#let logo_path = logo_path.replace("\\", "")
```
This seems overly complicated, and I feel like there might be a more efficient way to handle this.

Additionally, my `typst-show.typ` file is almost empty because I’m fetching all YAML parameters directly in `typst-template.typ`. My understanding, though, is that all parameters/variables should ideally be defined or manipulated in `typst-show.typ` and then made available in `typst-template.typ`.

A minimal example can be found here: <https://github.com/produnis/minimal_examples/tree/typstde>

Any suggestions on simplifying the logo path handling or better structuring the YAML parameter usage would be greatly appreciated! :-)

**Comment by gordonwoodhull** [comment]
Yes, as I understand it the escaping is a limitation of Pandoc in that it processes metadata as markdown. I have places where I need to pass raw Typst code via metadata,  and I ended up removing the backslashes and, even worse, running `eval` on the Typst side. 😱 

But it works. 

Best I can suggest is declaring a function to do the unescaping and then it can be done in one line in `typst-show.typ`.  Like this, only hopefully you don't need the eval:

https://github.com/quarto-dev/quarto-cli/blob/adc7186f0ee20ed125588cd603330beb2e76bc71/src/resources/formats/typst/pandoc/quarto/typst-show.typ#L62

**Reply by cscheid** [reply]
FWIW, there's a technique we use internally that is (partly) applicable here. A raw inline or block can be declared as `pandoc-native`, and Quarto will send the contents of that element through Pandoc's `native` reader. As a result

```
---
key1: '`Str "no--en--dashes--please"`{=pandoc-native}'
key2: yes--en--dashes--please
---

{{< meta key1 >}}

{{< meta key2 >}}
```

<img width="334" alt="image" src="https://github.com/user-attachments/assets/81552daa-17b3-4879-8d72-cdbea4305154">

This doesn't help end users much, because the syntax is atrociously ugly to teach. But @gordonwoodhull you might be able to use this. @produnis I don't know how much syntax pain you're willing to tolerate...

What we should 100% have is a mechanism for some metadata to declare itself as "never-markdown", and then we convert those entries ahead of sending them to Pandoc.

**Reply by gordonwoodhull** [reply]
In my case I'm generating metadata in Lua, so I was able to mark the attribute value as raw Typst directly:

```lua
          color = pandoc.RawInline('typst', color)
```

Much nicer, no evil eval. See 5e37696347

**Comment by produnis** [comment]
Today, archlinux updated to quarto 1.6.37,
and if you declared a `cls`  in your `YAML`, quarto will insert a `#set bibliography(style: csl)` without giving you the chance to  `#let csl = csl.replace("\\", "")`, leaving you with wrong "`\`" inside the path.

My workaround is to declare my own variable, like `mycsl`. This way, quarto won't trigger a `#set bibliography()` and I can replace the "`\\`" and call `#set bibliography(style: mycsl)` on my own.

**Reply by cscheid** [reply]
Could you give us a complete reprex? We'll open an issue.

**Reply by produnis** [reply]
Hi there, I uploaded my example to <https://github.com/produnis/minimal_examples/tree/quartotypstcsl>.

In `_extensions/hsnr-article/_extension.yml` I set the default `csl` to `_extensions/hsnr-article/apa-single-spaced.csl`.

In `typst-template.typ`, I tried to replace it in the first lines... I commented my `#set bibliogrpahy` call to show that quarto adds its own call...

**Reply by cscheid** [reply]
Thanks. Can you find an example that doesn't go through an extension?

**Reply by produnis** [reply]
sure, a "real" minimal example without any extension is here: <https://github.com/produnis/minimal_examples/tree/quartotypstcsl2> (with a **2** at the end)

**Reply by cscheid** [reply]
Thank you! Tracking the regression here: https://github.com/quarto-dev/quarto-cli/issues/11552
